src/FormatMetadata.cpp

```diff
@@ -0,0 +1,57 @@
+// Chemfiles, a modern library for chemistry file reading and writing
+// Copyright (C) Guillaume Fraux and contributors -- BSD license
+
+#include "chemfiles/FormatMetadata.hpp"
+
+#include "chemfiles/utils.hpp"
+#include "chemfiles/error_fmt.hpp"
+using namespace chemfiles;
+
+static void check_not_empty(const char* value, string_view field, const char* name) {
+    if (std::string(value).empty()) {
+        auto for_name = name == nullptr || field == "name" ? "" : fmt::format(" for '{}'", name);
+        throw format_error("the format {} can not be an empty string{}", field, for_name);
```

Comment:
I fell the possessive form "format's" would be clearer, because currently one might misunderstand and expect the name of the format e.g. "the format XYZ ..." when skimming the error messages

Reply:
I went with `the XXX can not be YYY for format ZZZ`, do you find it better?